ethstore/src/ethstore.rs

```diff
@@ -350,6 +382,54 @@ impl SimpleSecretStore for EthMultiStore {
 		self.import(vault, account)
 	}
 
+	fn insert_derived(&self, vault: SecretVaultRef, account_ref: &StoreAccountRef, password: &str, derivation: Derivation)
+		-> Result<StoreAccountRef, Error>
+	{
+		let accounts = self.get(account_ref)?;
+		for account in accounts {
+			// Skip if password is invalid
+			if !account.check_password(password) {
+				continue;
+			}
+			let extended = self.generate(account.crypto.secret(password)?, derivation)?;
+			return self.insert_account(vault, extended.secret().secret().clone(), password);
+		}
+		Err(Error::InvalidPassword)
+	}
+
+	fn generate_derived(&self, account_ref: &StoreAccountRef, password: &str, derivation: Derivation)
+	    -> Result<Address, Error>
+	{
+		let accounts = self.get(&account_ref)?;
+		for account in accounts {
+			// Skip if password is invalid
+			if !account.check_password(password) {
+				continue;
+			}
+			let extended = self.generate(account.crypto.secret(password)?, derivation)?;
+
+			return Ok(ethkey::public_to_address(extended.public().public()));
+		}
+		Err(Error::InvalidPassword)
+	}
+
+	fn sign_derived(&self, account_ref: &StoreAccountRef, password: &str, derivation: Derivation, message: &Message)
+		-> Result<Signature, Error>
+	{
+		let accounts = self.get(&account_ref)?;
+		for account in accounts {
+			// Skip if password is invalid
+			if !account.check_password(password) {
+				continue;
+			}
+			let extended = self.generate(account.crypto.secret(password)?, derivation)?;
+			let secret = extended.secret().secret();
```

Comment:
well it's not that weird
first `secret()` is `ExtendedSecret` of the extended key pair and second `secret()` is a raw `H256` secret of this extended secret, i don't see logical reason to `Deref` them in something they are not in any way equal to